src/com/amazon/ion/impl/IonReaderContinuableCore.java

```diff
@@ -0,0 +1,253 @@
+package com.amazon.ion.impl;
+
+import com.amazon.ion.Decimal;
+import com.amazon.ion.IntegerSize;
+import com.amazon.ion.IonCursor;
+import com.amazon.ion.IonInt;
+import com.amazon.ion.IonType;
+import com.amazon.ion.IvmNotificationConsumer;
+import com.amazon.ion.Timestamp;
+import com.amazon.ion.UnknownSymbolException;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.Date;
+
+/**
+ * IonCursor with the core IonReader interface methods. Useful for adapting an IonCursor implementation into a
+ * system-level IonReader.
+ */
+interface IonReaderContinuableCore extends IonCursor {
```

Comment:
There's a lot in here about the current value, current position, but even though tracking this state is vitally important to the function of the reader/cursor there's no way here to surface it to the consumer. 

Is this an appropriate layer to worry about that? While we can't at this level of generality say a lot about what "position" means, there are patterns for dealing with that kind of ambiguity- as when in practice you have to construct a [GarbageCollectionNotificationInfo](https://docs.oracle.com/javase/8/docs/jre/api/management/extension/com/sun/management/GarbageCollectionNotificationInfo.html) from a [CompositeData](https://docs.oracle.com/javase/8/docs/api/javax/management/openmbean/CompositeData.html) in order to learn about garbage collection information from within the runtime.